src/components/Modals/EditSemester.vue

```diff
@@ -0,0 +1,220 @@
+<template>
+  <div class="editSemesterModal">
+    <div class="editSemesterModal-content" id="deleteSemester">
+      <div class="editSemesterModal-top">
+        <span class="editSemesterModal-title">{{ title }}</span>
+        <img class="editSemesterModal-exit" src="@/assets/images/x.png" @click="closeCurrentModal" />
+      </div>
+      <div class="editSemesterModal-body">
+        <newSemester
+            class="modal-body"
+            :currentSemesters="semesters"
+            :id="deleteSemID"
+            :isEdit=true
+            ref="modalBodyComponent">
+        </newSemester>
+      </div>
+      <div class="editSemesterModal-buttonWrapper">
+        <button class="editSemesterModal-button" @click="closeCurrentModal">{{ cancel }}</button>
+        <div class="editSemesterModal-button editSemesterModal-button--delete" @click="editSemester">
+            <div class="editSemesterModal-button-left">
+                <span class="editSemesterModal-button-left-text">Edit</span>
+            </div>
+        </div>
+      </div>
+    </div>
+  </div>
+</template>
+
+<script>
+import Vue from 'vue';
+import NewCourse from '@/components/Modals/NewCourse';
+import NewCustomCourse from '@/components/Modals/NewCustomCourse';
+import NewSemester from '@/components/Modals/NewSemester';
+
+Vue.component('newCourse', NewCourse);
+Vue.component('newCustomCourse', NewCustomCourse);
+Vue.component('newSemester', NewSemester);
+
+export default {
+  data() {
+    return {
+      seasonInput: 'winter',
+      yearInput: 2020
+    };
+  },
+  props: {
+    semesters: Array,
+    deleteSemID: Number,
+    deleteSemType: String,
+    deleteSemYear: Number
+  },
+
+  computed: {
+    text() {
+      return 'Are you sure you want to edit this semester?';
+    },
+    cancel() {
+      return 'Cancel';
+    },
+    title() {
+      return 'Edit Semester';
+    }
+  },
+  methods: {
+    closeCurrentModal() {
+      const modal = document.getElementById(`editSemesterModal-${this.deleteSemID}`);
```

Comment:
DOM manipulation here is not great. You do not need to fix it because this is just duplicated from my old code (that I will fix soon), but try not to reproduce this in the future.